omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentRestController1_10Test.java

```diff
@@ -281,6 +285,57 @@ public void postAttachment_shouldUploadFileToEncounter() throws Exception {
 		Assert.assertEquals(obs.getEncounter().getUuid(), encounter.getUuid());
 	}
 	
+	@Test
+	public void postAttachment_shouldAcceptBase64Content() throws Exception {
+		String fileCaption = "Test file caption";
+		String fileName = "testFile2.dat";
+		String base64Content = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z8BQDwAEhQGAhKmMIQAAAABJRU5ErkJggg==";
+		Patient patient = Context.getPatientService().getPatient(2);
+		
+		MockMultipartHttpServletRequest request = newUploadRequest(getURI());
+		MockMultipartFile file = new MockMultipartFile("file", fileName, "application/octet-stream", randomData);
+		
+		request.addFile(file);
+		request.addParameter("patient", patient.getUuid());
+		request.addParameter("fileCaption", fileCaption);
+		request.addParameter("base64Content", base64Content);
+		
+		// Replay
+		SimpleObject response = deserialize(handle(request));
+		
+		Obs obs = Context.getObsService().getObsByUuid((String) response.get("uuid"));
+		Obs complexObs = Context.getObsService().getComplexObs(obs.getObsId(), null);
+		ComplexData complexData = complexObs.getComplexData();
```

Comment:
Now that you've got the complex data of the saved obs, you can get the bytes right away with [this](https://github.com/openmrs/openmrs-module-attachments/blob/91a848d462c989b0d3669f63d4a37eb2df9c6cff/api/src/main/java/org/openmrs/module/attachments/obs/BaseComplexData.java#L36) util method. And then you could encode this as Base64 and assert that you got back the same as `base64Content`.

Reply:
For some reason the encodings are not the same. When I used bufferedimage, the pixels matched but when I try to encode the `byte[]` to a base64 string, they're not equivalent. 

![Screen Shot 2020-06-22 at 14 19 09](https://user-images.githubusercontent.com/11330494/85292189-6ec7c700-b493-11ea-8bc1-df7452947d8b.png)

Reply:
The bytes returned by `BaseComplexData.getByteArray(complexData);` doesn't match that returned by `Base64.decodeBase64(base64Conten);`

Reply:
The bytes returned by `BaseComplexData.getByteArray(complexData);` does not match that returned by `Base64.decodeBase64(base64Content.split(",")[1].trim().getBytes());`